commit/plugin_e2e_test.go

```diff
@@ -417,6 +416,208 @@ func TestPlugin_E2E_AllNodesAgree_TokenPrices(t *testing.T) {
 	}
 }
 
+func TestPlugin_E2E_AllNodesAgree_ChainFee(t *testing.T) {
+	params := defaultNodeParams(t)
+	merkleOutcome := baseMerkleOutcome(params.rmnReportCfg)
+	nodes := make([]ocr3types.ReportingPlugin[[]byte], len(oracleIDs))
+
+	newFeeComponents, newNativePrice, packedGasPrice := newRandomFees()
+	expectedChainFeeOutcome := chainfee.Outcome{
+		GasPrices: []ccipocr3.GasPriceChain{
+			{
+				GasPrice: packedGasPrice,
+				ChainSel: destChain,
+			},
+		},
+	}
+
+	newFeeComponents2, newNativePrice2, newPackedGasPrice2 := newRandomFees()
+
+	testCases := []struct {
+		name                  string
+		prevOutcome           Outcome
+		expOutcome            Outcome
+		expTransmittedReports []ccipocr3.CommitPluginReport
+
+		mockCCIPReader func(*readerpkg_mock.MockCCIPReader)
+	}{
+		{
+			name:        "fee components should be updated",
+			prevOutcome: Outcome{},
+			expOutcome: Outcome{
+				MerkleRootOutcome: merkleOutcome,
+				ChainFeeOutcome: chainfee.Outcome{
+					GasPrices: []ccipocr3.GasPriceChain{
+						{
+							GasPrice: packedGasPrice,
+							ChainSel: destChain,
+						},
+						{
+							GasPrice: packedGasPrice,
+							ChainSel: sourceChain1,
+						},
+						{
+							GasPrice: packedGasPrice,
+							ChainSel: sourceChain2,
+						},
+					},
+				},
+			},
+			mockCCIPReader: func(m *readerpkg_mock.MockCCIPReader) {
+				m.EXPECT().GetAvailableChainsFeeComponents(params.ctx, mock.Anything).Unset()
```

Comment:
Could you move the `Unset()`'s to the main test loop along with a comment about why that the default test setup doesn't apply to this test.